### PR TITLE
[GITHUB-1] Prometheus PHP FPM exporter

### PR DESCRIPTION
**Pull request by rsgoodall**

### CODE REVIEW

File: defaults/main.yml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+---
+
+prometheus_phpfpm_exporter:
+  checksum: sha256:db234e6eead69a1ffad77137392452dfa421d03e60ff3c5211c8ab72bf514648
+  download_dest: "/tmp"
+  download_url: "https://github.com/bakins/php-fpm-exporter/releases/download/v0.1.1/php-fpm-exporter.linux.amd64"
+  group: prometheus
+  opts:
+#  opts: --addr 0.0.0.0:9238 --endpoint http://127.0.0.1:80/status
```
Comment (racer1680) [inline defaults/main.yml]: Do you need comment here?

File: tests/test.yml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+        - build
+
+  roles:
+    - role: sansible.prometheus_phpfpm_exporter
+
+  post_tasks:
+    - name: Get Prometheus PHP FPM Exporter service state
+      become: yes
+      service:
+        name: prometheus-phpfpm-exporter
```
Comment (racer1680) [inline tests/test.yml:22]: I thought we use "_" in services' names...

File: tasks/build.yml [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    url: "{{ prometheus_phpfpm_exporter.download_url }}"
+    dest: "{{ prometheus_phpfpm_exporter.path.install }}/prometheus_phpfpm_exporter/phpfpm_exporter"
+    checksum: "{{ prometheus_phpfpm_exporter.checksum }}"
+    owner: "{{ prometheus_phpfpm_exporter.user }}"
+    group: "{{ prometheus_phpfpm_exporter.group }}"
+    mode: 0755
+
+- name: Set Prometheus PHP FPM Exporter defaults
+  become: yes
+  lineinfile:
```
Comment (racer1680) [inline tasks/build.yml]: Why not template?

Comment (lobsterdore) [reply tasks/build.yml]: Agreed, I think a template would be better here.

Comment (lobsterdore) [reply tasks/build.yml]: And since it's a template it will be need to be created during the configure stage.

File: tests/test.yml (hunk 2 above)
Comment (lobsterdore) [reply tests/test.yml:22]: We don't have a standard for service names, he's used hyphens in the other roles so maybe we should stick to that for Prometheus.

File: tasks/build.yml [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+    src: prometheus_exporter.sh
+    dest: "{{ prometheus_phpfpm_exporter.path.install }}/prometheus_exporter.sh"
+    owner: "{{ prometheus_phpfpm_exporter.user }}"
+    group: "{{ prometheus_phpfpm_exporter.group }}"
+    mode: 0755
+
+- name: Create Prometheus PHP FPM Exporter directory
+  become: yes
+  file:
+    path: "{{ prometheus_phpfpm_exporter.path.install }}/prometheus_phpfpm_exporter"
```
Comment (lobsterdore) [inline tasks/build.yml]: Can we add this dir to the 'Create Prometheus PHP FPM Exporter directories' task? Then you can get rid of this task.

File: tests/test.yml (hunk 2 above)
Comment (rsgoodall) [reply tests/test.yml:22]: I think hyphen is more standard for linux / unix init scripts.

**Comment by lobsterdore** [comment]
LGTM 👍

File: tests/test.yml (hunk 2 above)
Comment (lobsterdore) [reply tests/test.yml:22]: OK, hyphens it is then

**Review by lobsterdore — COMMENTED** [review]
Approved!

**Review by lobsterdore — APPROVED** [review]

**Pull request merged**